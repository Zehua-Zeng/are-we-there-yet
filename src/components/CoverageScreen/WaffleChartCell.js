import React from "react";
// styled components
import styled from 'styled-components';

const StyledChartCellContainer = styled.div`
    .grid-values {
        margin: 0.2rem;
        width: 1.2rem;
        height: 1.2rem;
        font-size: 1rem;
        cursor: default;
        position: relative;

        .tooltiptext {
            visibility: hidden;
            width: 125px;
            background-color: rgba(0,0,0,0.8);
            color: #fff;
            text-align: center;
            padding: 5px 0;
            border-radius: 6px;
            font-size: 0.8rem;
            
            position: absolute;
            top: -1rem;
            left: 2rem; 
            z-index: 2;

            p {
                margin: 0;
            }

            .tooltip-title {
                font-weight: 500;
                text-align: start;
                padding: 0.2rem 0.5rem;
            }

            .tooltip-tag {
                width: fit-content;
                padding: 0.2rem;
                border-radius: 0.2rem;
                margin: 0 auto 0.2rem;
                color: black;
                font-size: 0.5rem;
            }

            .tag-design {
                background-color: lightgray;
                color: black;
            }

            .tag-hybrid {
                /* color: rgb(181, 237, 152); */
                background-color: rgb(181, 237, 152);
            }

            .tag-experiment {
                background-color: lightblue;
            }

            .tag-theory {
                background-color: rgb(250, 224, 152);
            }
        }

        .tooltiptext::after {
            content: " ";
            position: absolute;
            top: 20%;
            right: 100%; /* To the left of the tooltip */
            margin-top: -5px;
            border-width: 5px;
            border-style: solid;
            border-color: transparent black transparent transparent;
        }
    }

    .grid-values:hover .tooltiptext {
        visibility: visible
    }

    .experiment-deactive {
        color: lightblue;
        background-color: lightblue;
    }

    .experiment-deactive:hover {
        color: #70C2E8;
        background-color: #70C2E8;
    }

    .hybrid-deactive {
        color: rgb(181, 237, 152);
        background-color: rgb(181, 237, 152);
    }

    .hybrid-deactive:hover {
        color: rgb(162, 222, 131);
        background-color: rgb(162, 222, 131);
    }

    .theory-deactive {
        color: rgb(250, 224, 152);
        background-color: rgb(250, 224, 152);
    }

    .theory-deactive:hover {
        color: rgb(245, 210, 118);
        background-color: rgb(245, 210, 118);
    }

`;

var WaffleChartCell = ({category, name}) => {
    return (
        <StyledChartCellContainer key={name} data-testid={name}>
            <div className={`grid-values ${category.toLowerCase()+"-deactive"}`} >
                <span className="tooltiptext">
                    <div>
                        <p className="tooltip-title">Coverage:</p>
                        <p className={`tooltip-tag tag-${category.toLowerCase()}`}>        {category.toLowerCase()} 
                        </p>
                    </div>
                    <div>
                        <p className="tooltip-title">Encoding Design:</p>
                        <p className={`tooltip-tag tag-design`}> 
                            {name}
                        </p>
                    </div>
                </span>
            </div>
        </StyledChartCellContainer>
    );
}

export default WaffleChartCell;