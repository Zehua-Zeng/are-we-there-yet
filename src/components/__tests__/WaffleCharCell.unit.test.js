import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
// Do not remove, used in the file
import {toBeInTheDocument} from '@testing-library/jest-dom';
import WaffleChartCell from '../CoverageScreen/WaffleChartCell';

afterEach(() => {
    cleanup();
});

test('should render experiemnt cell matches snapshot', () => {
    let name= "PXPY";
    let category = "Experiment";

    let cell = <WaffleChartCell name={name} category={category} key={name}/>;
    render(cell);

    const cellElement = screen.getByTestId(name);
    // expect it to be rendered in DOM
    expect(cellElement).toBeInTheDocument();

    const tree = renderer.create(cell).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should render theory cell matches snapshot', () => {
    let name= "PXPY";
    let category = "Theory";

    let cell = <WaffleChartCell name={name} category={category} key={name}/>;
    render(cell);

    const cellElement = screen.getByTestId(name);
    // expect it to be rendered in DOM
    expect(cellElement).toBeInTheDocument();

    const tree = renderer.create(cell).toJSON();
    expect(tree).toMatchSnapshot();
});

test('should render hybrid cell matches snapshot', () => {
    let name= "PXPY";
    let category = "Hybrid";

    let cell = <WaffleChartCell name={name} category={category} key={name}/>;
    render(cell);

    const cellElement = screen.getByTestId(name);
    // expect it to be rendered in DOM
    expect(cellElement).toBeInTheDocument();

    const tree = renderer.create(cell).toJSON();
    expect(tree).toMatchSnapshot();
});