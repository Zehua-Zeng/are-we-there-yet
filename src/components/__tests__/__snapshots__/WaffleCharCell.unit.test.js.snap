// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render experiemnt cell matches snapshot 1`] = `
<div
  className="sc-bdnxRM kYbNFG"
  data-testid="PXPY"
>
  <div
    className="grid-values experiment-deactive"
  >
    <span
      className="tooltiptext"
    >
      <div>
        <p
          className="tooltip-title"
        >
          Coverage:
        </p>
        <p
          className="tooltip-tag tag-experiment"
        >
                  
          experiment
        </p>
      </div>
      <div>
        <p
          className="tooltip-title"
        >
          Encoding Design:
        </p>
        <p
          className="tooltip-tag tag-design"
        >
          PXPY
        </p>
      </div>
    </span>
  </div>
</div>
`;

exports[`should render hybrid cell matches snapshot 1`] = `
<div
  className="sc-bdnxRM kYbNFG"
  data-testid="PXPY"
>
  <div
    className="grid-values hybrid-deactive"
  >
    <span
      className="tooltiptext"
    >
      <div>
        <p
          className="tooltip-title"
        >
          Coverage:
        </p>
        <p
          className="tooltip-tag tag-hybrid"
        >
                  
          hybrid
        </p>
      </div>
      <div>
        <p
          className="tooltip-title"
        >
          Encoding Design:
        </p>
        <p
          className="tooltip-tag tag-design"
        >
          PXPY
        </p>
      </div>
    </span>
  </div>
</div>
`;

exports[`should render theory cell matches snapshot 1`] = `
<div
  className="sc-bdnxRM kYbNFG"
  data-testid="PXPY"
>
  <div
    className="grid-values theory-deactive"
  >
    <span
      className="tooltiptext"
    >
      <div>
        <p
          className="tooltip-title"
        >
          Coverage:
        </p>
        <p
          className="tooltip-tag tag-theory"
        >
                  
          theory
        </p>
      </div>
      <div>
        <p
          className="tooltip-title"
        >
          Encoding Design:
        </p>
        <p
          className="tooltip-tag tag-design"
        >
          PXPY
        </p>
      </div>
    </span>
  </div>
</div>
`;
